### PR TITLE
Changes following feedback from Sonarcloud issue log. 

### PR DESCRIPTION
**Pull request by iamianmullins**
Creation of additional functions to reduce code complexity and improve readability.
Let me know your thoughts.
Thanks Eric

### CODE REVIEW

File: src/drm.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <memory>
```
Comment (ericcurtin) [inline src/drm.h:14]: Looks great, everything could go through git clang-format, well above 80 characters

File: src/kms_sink.cpp [hunk 2]
```diff
@@ -33,34 +33,45 @@ KMSSink::KMSSink(const std::string& connectorName) {
    * pick the first connected connector or, if no connector is connected,
```
Comment (ericcurtin) [inline src/kms_sink.cpp:33]: Looks great

File: src/kms_sink.h [hunk 3]
```diff
@@ -47,6 +47,7 @@ class KMSSink {
   int selectPipeline(const libcamera::PixelFormat& format);
```
Comment (ericcurtin) [inline src/kms_sink.h:47]: Looks great

File: src/drm.cpp [hunk 4]
```diff
@@ -469,16 +462,39 @@ int Device::getResources() {
     objects_[obj.id()] = &obj;
   }
 
-  /* Finally, perform all delayed setup of mode objects. */
+  performAllDelayedSetupOfModeObjects(objects_);
```
Comment (ericcurtin) [inline src/drm.cpp:465]: The old code did not return zero in all cases:

`return performAllDelayedSetupOfModeObjects(objects_);`

would work though.

Comment (iamianmullins) [reply src/drm.cpp:465]: Thanks for the feedback Eric, I'll change this

**Comment by ericcurtin** [comment]
Merged

**Pull request closed without merging**